### ISSUE TITLE
UPower module

### ISSUE THREAD

**Issue by cyrinux** [lasers/py3status#3]
With upower we can have now more information about battery in with more modern way.
With can use parsing of 'upower --enumerate/dump', or use dbus.

My first goal was to have information about my battery mouse (keyboard in a same way but i have not one).
I have logitech mice, and see information for those i /sys/class/power, acpi, upower (dbus upower) but only upower seems revelent and usable.

I tried battery_level, but 'sys' methode don't parse 'hid*' device, and acpi report wrong information.

Feel free to discuss before i start something.

**Comment by lasers** [comment]
https://github.com/ultrabug/py3status/pull/1084

**Comment by lasers** [comment]
That PR supports only the battery for `battery_module` and does not display other items. I could see how it is useful to make `upower` module displaying a list of devices.

If you want to make it, that would be cool. I have made many PRs and modules that didn't go in `py3status`.... So no worries.

This, I see, could be useful for laptops. We probably could add a `filter` config too.

**Comment by lasers** [comment]
I revisit this again. The usage for this module may be very low. Look at `lm_sensors` PR if you need help with parsing some of `upower` information. I'm hoping to see py3 dbus helper someday as this could give us more interesting things to work with. We can get it at `org.freedesktop.UPower` too.

**Comment by lasers** [comment]
Is this logitech unifying mouse? You could try this https://github.com/ultrabug/py3status/pull/919

**Comment by cyrinux** [comment]
Seems exactly what I would like to do. I allready use solaar and will try
this one. Thanks man

**Comment by lasers** [comment]
Usable `upower -d` placeholders.
```
Laptop battery.             Logitech Unifying devices.
* state: fully-charge       * state: unknown   # not connected)
* percentage: 94%           * state: discharging
* capacity: 67.3396%    :(  * percentage: 10%    :(
```
For me, the updates are really slow with Logitech Unifying devices... with `(66 seconds ago)` or such in `updated:` line. My Logitech KB/mouse were moving around for more than 20 seconds and there were no new changes from `upower -d`. Other than `battery`, I'm not sure we'll get lot of usage out of this. Maybe it's different for you?

It'd be fun to update `logitech_unifying` module. I looked at it again. I did some wrong shit in there with `thresholds`. Could remove `is_toggled` boolean in favor of `format` and/or storage value so it stays toggled off... or we can just stay with `solaar` tray. Anyway, I'm closing this. :-)
